src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/writer/CatalogWriter.java

```diff
@@ -0,0 +1,152 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2016 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.backuprestore.writer;
+
+import org.geoserver.backuprestore.AbstractExecutionAdapter;
+import org.geoserver.backuprestore.Backup;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.config.util.XStreamPersister;
+import org.geoserver.config.util.XStreamPersisterFactory;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.annotation.BeforeStep;
+import org.springframework.batch.item.ItemReader;
+import org.springframework.batch.item.file.ResourceAwareItemWriterItemStream;
+import org.springframework.batch.item.support.AbstractItemStreamItemWriter;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+
+import com.thoughtworks.xstream.XStream;
+
+/**
+ * Abstract Spring Batch {@link ItemReader}.
+ * 
+ * Configures the {@link Catalog} and initizializes the {@link XStreamPersister}.
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+public abstract class CatalogWriter<T> extends AbstractItemStreamItemWriter<T>
+        implements ResourceAwareItemWriterItemStream<T>, InitializingBean {
+    
+    protected Class clazz;
+
+    protected Backup backupFacade;
+
+    protected Catalog catalog;
+
+    protected XStreamPersister xstream;
+
+    private XStream xp;
+    
+    private boolean isNew;
+
+    private AbstractExecutionAdapter currentJobExecution;
+
+    private boolean dryRun;
+
+    private boolean bestEffort;
+
+    private XStreamPersisterFactory xStreamPersisterFactory;
+    
+    public CatalogWriter(Class<T> clazz, Backup backupFacade,
+            XStreamPersisterFactory xStreamPersisterFactory) {
+        this.clazz = clazz;
+        this.backupFacade = backupFacade;
+        this.xStreamPersisterFactory = xStreamPersisterFactory;
+
+        this.setExecutionContextName(ClassUtils.getShortName(clazz));
+    }
+
+    @BeforeStep
+    protected void retrieveInterstepData(StepExecution stepExecution) {
+        // Accordingly to the running execution type (Backup or Restore) we
+        // need to validate resources against the official GeoServer Catalog (Backup)
+        // or the temporary one (Restore).
+        //
+        // For restore operations the order matters.
+        JobExecution jobExecution = stepExecution.getJobExecution();
```

Comment:
Besides the observation that the backup context might be a better place to fetch the info, this block of code is really repeated over and over and over.... consider building a support class that does it and stores the results, or a base class or... something? :-)